src/transaction/log_applier.c

```diff
@@ -2962,7 +2962,7 @@ la_add_apply_list (int tranid)
  *   log_pageid, log_offset, and log_pgptr are set as a side effect.
  */
 static void
-la_log_copy_fromlog (char *rec_type, char *area, int length, LOG_PAGEID log_pageid, PGLENGTH log_offset,
+la_log_copy_fromlog (char *rec_type, char *area, int &length, LOG_PAGEID log_pageid, PGLENGTH log_offset,
```

Comment:
How does this solve the problem? Could you elaborate on it more?

2993L: `length = length - DB_SIZEOF (INT16);` solve the problem? and is it always done? 
What happens if `rec_type` is NULL?

Reply:
Whether rec_type is null or not is a kind of condition. If rec_type is null, the size of length passed is read from the log page. If rec_type is not null, rec_type with a length of 2 bytes is read, and then length - 2 is read. In both cases, the read length is the same, but the behavior differs.

The la_log_copy_fromlog() function seems to serve several purposes. It is used to read the format like | rec_type | recdes->data | (redo image) in the case of this issue, but it can also be used to read record headers, such as LOG_REC_REPLICATION.

Reply:
I guess the modified length would be used to determine its type in salve node from your code and explanation. Correct? I don't really know, but I'm worried some of callers of `la_log_copy_fromlog() just hand over a local variable for the parameter `length`. Don't these cases matter? Which caller is the case causes the issue? If you have already checked, it's OK.

Reply:
The following is the callstack you want.

0 la_log_copy_fromlog()
1 la_get_log_data()
2 la_get_recdes()
3 la_apply_insert_log()
...